src/Controller/ComponentRegistry.php

```diff
@@ -147,6 +147,7 @@ protected function _create(object|string $class, string $alias, array $config):
             /** @var \Cake\Controller\Component $instance */
             $instance = $this->container->get($class);
             $instance->setConfig($config);
+            $instance->initialize($config);
```

Comment:
Extending the container definition to include `$config` is a better solution imo.